presto-main/src/main/java/io/prestosql/execution/SqlQueryExecution.java

```diff
@@ -604,7 +604,7 @@ private boolean shouldWaitForMinWorkers(Statement statement)
         private final SubPlan root;
         private final boolean summarizeTaskInfos;
 
-        public PlanRoot(SubPlan root, boolean summarizeTaskInfos)
```

Comment:
Same as above. The class is private, but its methods are part of the public interface of the PlanRoot class, so they should remain public.

The litmus test for this is: what if you wanted to move the class one level up? Aside from changing the class visibility modifier, it should just work.

Reply:
This is a matter of taste. And we have been in this discussion already several times. IDE thinks it is a warning, typically IDE is correct. I won't argue here and I will apply this comment.

When I move class, then the design changes and methods and their accesses should be revisited. Also it is safer to keep methods private, so something do not get accidentally public.

Besides that, do you have anything else

Reply:
>  IDE thinks it is a warning, typically IDE is correct

That's also a matter of what warnings you have enabled. It doesn't show as a warning for me.

> When I move class, then the design changes and methods and their accesses should be revisited. Also it is safer to keep methods private, so something do not get accidentally public.

On the other hand, using private methods in this case is more confusing. `private` for methods of inner classes don't behave in the same way as for top-level classes -- they are not truly private, as they are visible to the enclosing class. This creates some "cognitive dissonance" and requires one to look at the code carefully to understand whether the method is meant to be used outside of the inner class or not. Using `private` vs `public` convention to indicate this makes it easier to reason about the code.

> Besides that, do you have anything else

No, that was it.

Reply:
Either way is fine to me, I think we should write it down somewhere and so everyone else follow this and don't waste more time on it.

Reply:
https://github.com/prestosql/presto/wiki/Code-style :)